### ISSUE TITLE
No support for arrays with custom types

### ISSUE THREAD

**Issue by Aryan807** [denesfilotas/vscode-blitz3d#3]
The parser does not seem to support arrays of types

Example: the following code results in error

```
Type test
    Field testVar%
    Field testString$
    Field Var%[2]
End Type

Dim mytest.test(2)

mytest(2) = new test

mytest(2)\testVar = 2
mytest(2)\testString = "Test String"

Print(mytest(2)\testVar)
Print(mytest(2)\testString)
```

**Comment by denesfilotas** [comment]
Hi!

Thanks for the issue. I rarely use arrays with dim, so I didn't really notice the bug. Looks like it was an easy fix.

The update should be available on vscode marketplace. Feel free to let me know if there are any more bugs like this.

ps I also noticed that if mytest was a function that returns a variable with type test, it would not possible to use the same simple expression syntax, e.g. `Print(mytest(2)\testString)` would result in a syntax error. Interesting!

**Comment by Aryan807** [comment]
Thanks for the quick reponse.

Unfortunately, our project still doesn't work.
It seems I have found another issue: nested Includes don't work.

Say for example I made a file "test.bb" referencing "folder/test1.bb" in a folder "folder" and the latter file referencing "folder/test2.bb" then I get error in "folder/test1.bb" that test2.bb was not found. "test.bb" works fine though.

![image](https://github.com/denesfilotas/vscode-blitz3d/assets/52038234/e380284d-7fd2-4af2-846f-f846bf54a698)

This issue was not present in v0.4.x releases
